src/test/java/org/lsc/Ldap2LdapHookSyncTest.java

```diff
@@ -0,0 +1,261 @@
+
+		// perform the sync
+		launchSyncCleanTask(sync_tasks, false, true, false);
+		launchSyncCleanTask(clean_tasks, false, false, true);
+
+		// check the results of the synchronization
+		reloadJndiConnections();
+
+		ObjectMapper mapperCreatedEntry = new ObjectMapper();
+		JsonNode expectedCreatedEntry = mapperCreatedEntry.readTree("[ { \"attributeName\" : \"objectClass\", \"values\" : [ \"inetOrgPerson\", \"person\", \"top\" ], \"operation\" : \"ADD_VALUES\" }, { \"attributeName\" : \"cn\", \"values\" : [ \"CN0001-hook\" ], \"operation\" : \"ADD_VALUES\" }, { \"attributeName\" : \"sn\", \"values\" : [ \"CN0001-hook\" ], \"operation\" : \"ADD_VALUES\" } ]");
+
+		checkJSONSyncResults("create", expectedCreatedEntry);
+
+		ObjectMapper mapperUpdatedEntry = new ObjectMapper();
+		JsonNode expectedUpdatedEntry = mapperUpdatedEntry.readTree("[ { \"attributeName\" : \"description\", \"values\" : [ \"CN0001-hook\" ], \"operation\" : \"REPLACE_VALUES\" } ]");
+
+		checkJSONSyncResults("update", expectedUpdatedEntry);
+
+		checkJSONSyncResults("delete", null);
+	}
+
+	@Test
+	public final void testLdap2LdapLDIFHookSyncTest() throws Exception {
+
+		// Declare the tasks to launch in the correct order
+		List<String> sync_tasks = Arrays.asList("ldap2ldapLDIFHookTestCreate", "ldap2ldapLDIFHookTestUpdate");
+		List<String> clean_tasks = Arrays.asList("ldap2ldapLDIFHookTestDelete");
+
+		// perform the sync
+		launchSyncCleanTask(sync_tasks, false, true, false);
+		launchSyncCleanTask(clean_tasks, false, false, true);
+
+		// check the results of the synchronization
+		reloadJndiConnections();
+
+		List<String> expectedCreatedEntry = Arrays.asList(
+			"dn: cn=CN0001-hook,ou=ldap2ldap2TestTaskDst,ou=Test Data,dc=lsc-project,dc=org",
+			"changetype: add",
+			"objectClass: inetOrgPerson",
+			"objectClass: person",
+			"objectClass: top",
+			"cn: CN0001-hook",
+			"sn: CN0001-hook"
+		);
+
+		checkLDIFSyncResults("create", expectedCreatedEntry);
+
+		List<String> expectedUpdatedEntry = Arrays.asList(
+			"dn: cn=CN0001-hook,ou=ldap2ldap2TestTaskDst,ou=Test Data,dc=lsc-project,dc=org",
+			"changetype: modify",
+			"replace: description",
+			"description: CN0001-hook"
+		);
+
+		checkLDIFSyncResults("update", expectedUpdatedEntry);
+
+		checkLDIFSyncResults("delete", new ArrayList<String>());
+	}
+
+	/*
+	 * Read hook log file to check passed arguments and modification passed as input
+	*/
+	private final void checkLDIFSyncResults(String operation, List<String> expectedEntry) throws Exception {
+
+		List<String> hookResults = new ArrayList<String>();
+		try {
+			File hookFile = new File("hook-ldif-" + operation + ".log");
+			Scanner hookReader = new Scanner(hookFile);
+
+			while (hookReader.hasNextLine()) {
+				String data = hookReader.nextLine();
+				hookResults.add(data);
+			}
+			hookReader.close();
+			hookFile.delete(); // remove hook log
+		} catch (Exception e) {
+			fail("Error while reading hook-ldif-" + operation + ".log");
```

Comment:
you don't really need to try/catch in test, the test will fail anyway in case of unexpected exception.